src/components/instantlaunches/admin/InstantLaunchMapping.js

```diff
@@ -92,140 +92,141 @@ const initialValues = {
-                id={buildID(baseID, ids.NAME)}
-                name="mappingName"
-                label={t("common:name")}
-                className={classes.flexItem}
-                value={formik.values.mappingName}
-                onChange={formik.handleChange}
-                error={formik.touched.mappingName && formik.errors.mappingName}
-                helperText={
-                    formik.touched.mappingName && formik.errors.mappingName
-                }
-            />
-
-            <TextField
-                value={formik.values.patternKind}
-                onChange={formik.handleChange}
-                id={buildID(baseID, ids.PATTERN, ids.KIND, ids.MENU)}
-                name="patternKind"
-                label={t("patternKind")}
-                className={classes.flexItem}
-                select
-            >
-                <MenuItem
-                    value="glob"
-                    key="glob"
-                    id={buildID(
-                        baseID,
-                        ids.PATTERN,
-                        ids.KIND,
-                        ids.MENU,
-                        ids.GLOB
-                    )}
-                >
-                    {t("glob")}
-                </MenuItem>
-
-                <MenuItem
-                    value="infoType"
-                    key="infoType"
-                    id={buildID(
-                        baseID,
-                        ids.PATTERN,
-                        ids.KIND,
-                        ids.MENU,
-                        ids.INFO_TYPE
-                    )}
+        <Formik initialValues={initialValues} onSubmit={handleSubmit}>
+            {({ handleSubmit, values, handleChange, touched, errors }) => (
+                <form
+                    onSubmit={handleSubmit}
+                    className={classes.flexContainer}
+                    id={baseID}
                 >
-                    {t("infoType")}
-                </MenuItem>
-            </TextField>
-
-            {formik.values.patternKind === "infoType" ? (
-                <TextField
-                    id={buildID(baseID, ids.INFO_TYPE, ids.MENU)}
-                    name="pattern"
-                    label={t("infoType")}
-                    className={classes.flexItem}
-                    value={formik.values.pattern}
-                    onChange={formik.handleChange}
-                    select
-                >
-                    {infoTypes.map((infoType) => (
+                    <TextField
+                        id={buildID(baseID, ids.NAME)}
+                        name="mappingName"
+                        label={t("common:name")}
+                        className={classes.flexItem}
+                        value={values.mappingName}
+                        onChange={handleChange}
+                        error={touched.mappingName && errors.mappingName}
+                        helperText={touched.mappingName && errors.mappingName}
```

Comment:
All of these `value`, `onChange`, `error`, and `helperText` props can go away if this `TextField` is replaced by a formik `FastField` with a `component={FormTextField}` prop (and `FormTextField` is imported from `ui-lib`).

The `<FastField ... component={FormTextField}>` also handles `onBlur` events, so if a field is required, and the user enters it, then exits it without making any changes (i.e. if the field is "touched"), then it can display a `required` validation error message (though I don't see any required fields in this form).

So all of the `TextField` components in this form can be refactored the same way (even the `select` field).